### ISSUE TITLE
Update missing_device_type_components.py to check more than just names

### ISSUE THREAD

**Issue by ryanmerolle** [netbox-community/reports#77]
Update missing_device_type_components.py to check more than just names of items.

**Comment by ryanmerolle** [comment]
@candlerb I started with [this](https://github.com/ryanmerolle/reports/blob/issue_77/reports/dcim-reports/missing_device_type_components.py)

Thoughts?

**Comment by danner26** [comment]
Hey Ryan, just throwing this here as well since sometimes things get buried in slack.

Maybe you can build a hash table rather than using nested for loops? Data structures has never been my forte, but I do recall some of the math. Those 4 nested for loops are probably creating a lot of time complexity

**Comment by candlerb** [comment]
> Thoughts?

I think this should be a separate report, because there are two distinct things you might want to do:

* Check for device components which are *missing* according to the device type template
* Check for device components which are *different* from the device type template

That is: you may have intentionally changed something from the device type in a specific device, and don't want to be prompted about it all the time.

**Comment by ryanmerolle** [comment]
> That is: you may have intentionally changed something from the device type in a specific device, and don't want to be prompted about it all the time.

Good point. So a name focused Class like the original and a full detail Class.

**Comment by ryanmerolle** [comment]
Using @kkthxbye-code https://gist.github.com/kkthxbye-code/6d6f017e590d3271cbd1071fcb4840be as an example to refactor
